### ISSUE TITLE
C++ CodecService fails to encode non-top level entity to XML subtree

### ISSUE THREAD

**Issue by ygorelik** [CiscoDevNet/ydk-gen#1049]
## Current Behavior
When tried to get XML subtree from non-top level entity the CodecService fails with segmentation violation.

## Steps to Reproduce
Add the following unit test to the suite of bundle tests:
```
TEST_CASE("xml_codec_augment_subtree")
{
    CodecServiceProvider codec_provider{EncodingFormat::XML};
    CodecService codec_service{};

    auto passive = make_unique<ydktest_sanity::Runner::Passive>();
    passive->name = "xyz";

    auto ifc = make_shared<ydktest_sanity::Runner::Passive::Interfac>();
    ifc->test = "abc";
    passive->interfac.append(ifc);

    passive->testc->xyz = make_shared<ydktest_sanity::Runner::Passive::Testc::Xyz>();
    passive->testc->xyz->parent = passive.get();
    passive->testc->xyz->xyz = 25;

    auto xml = codec_service.encode(codec_provider, *passive, true, true);
    cout << xml << endl;
}
```
Run the unit test and observe the issue:
```
~/ydk-gen/sdk/cpp/tests/build$ ./ydk_bundle_test xml_codec_augment_subtree

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ydk_bundle_test is a Catch v1.7.1 host application.
Run with -? for options

-------------------------------------------------------------------------------
xml_codec_augment_subtree
-------------------------------------------------------------------------------
/home/ygorelik/ydk-gen/sdk/cpp/tests/test_xml_subtree.cpp:370
...............................................................................

/home/ygorelik/ydk-gen/sdk/cpp/tests/test_xml_subtree.cpp:370: FAILED:
due to a fatal error condition:
  SIGSEGV - Segmentation violation signal

===============================================================================
test cases: 1 | 1 failed
assertions: 1 | 1 failed

Segmentation fault (core dumped)
```

## System Information

YDK-0.8.5

**Comment by ygorelik** [comment]
#### Suggested workaround
Initialize root schema by creating connection to the device, then use XmlSubtreeCodec to get the XML subtree:
```
TEST_CASE("xml_codec_augment_subtree")
{
    ydk::path::Repository repo{TEST_HOME};
    ydk::path::NetconfSession session{repo, "127.0.0.1", "admin", "admin", 12022};
    XmlSubtreeCodec xml_codec{};

    auto passive = make_unique<ydktest_sanity::Runner::Passive>();
    passive->name = "xyz";

    auto ifc = make_shared<ydktest_sanity::Runner::Passive::Interfac>();
    ifc->test = "abc";
    passive->interfac.append(ifc);

    passive->testc->xyz = make_shared<ydktest_sanity::Runner::Passive::Testc::Xyz>();
    passive->testc->xyz->parent = passive.get();
    passive->testc->xyz->xyz = 25;

    auto xml = xml_codec.encode(*passive, session.get_root_schema());
    cout << xml << endl;
}
```
Run the test and observe correct results:
```
ygorelik@bionic:~/ydk-gen/sdk/cpp/tests/build$ ./ydk_bundle_test xml_codec_augment_subtree
<passive xmlns="http://cisco.com/ns/yang/ydktest-sanity">
  <name>xyz</name>
  <interfac>
    <test>abc</test>
  </interfac>
  <testc xmlns="http://cisco.com/ns/yang/ydktest-sanity-augm">
    <xyz>
      <xyz>25</xyz>
    </xyz>
  </testc>
</passive>
===============================================================================
test cases: 1 | 1 passed
assertions: - none -
```

**Comment by ygorelik** [comment]
Resolved the issue in my fork. It will be included to the next patch release - 0.8.5.3.
